ui/src/app/cohort-review/detail-tab-table/detail-tab-table.component.tsx

```diff
@@ -0,0 +1,277 @@
+        pageFilterType: this.props.filterType,
+        domain: this.props.domain,
+      } as PageFilterRequest;
+
+      cohortReviewApi().getParticipantData(
+        namespace,
+        id,
+        +cid,
+        +cdrVersionId,
+        this.props.participantId,
+        pageFilterRequest
+      ).then(response => {
+        this.setState({
+          data: response.items,
+          loading: false,
+        });
+      });
+    }
+
+    onSort = (event: any) => {
+      this.setState({sortField: event.sortField, sortOrder: event.sortOrder});
+    }
+
+    columnSort = (sortField: string) => {
+      if (this.state.sortField === sortField) {
+        const sortOrder = this.state.sortOrder === 1 ? -1 : 1;
+        this.setState({sortOrder});
+      } else {
+        this.setState({sortField, sortOrder: 1});
+      }
+    }
+
+    onPage = (event: any) => {
+      this.setState({start: event.first});
+    }
+
+    render() {
+      const {loading, rows, start, sortField, sortOrder} = this.state;
+      const data = this.state.data || [];
+      const lastRowOfPage = (start + rows) > data.length
+        ? start + rows - (start + rows - data.length) : start + rows;
+      const pageReportTemplate = (start + 1) + ' - ' + lastRowOfPage + ' of ' + data.length
+        + ' records ';
+      const paginatorTemplate = 'CurrentPageReport PrevPageLink PageLinks NextPageLink';
+
+      const columns = this.props.columns.map((col) => {
+        const asc = sortField === col.name && sortOrder === 1;
+        const desc = sortField === col.name && sortOrder === -1;
+        const header = <React.Fragment>
+          <span
+            onClick={() => this.columnSort(col.name)}
+            style={styles.columnHeader}>
+            {col.displayName}
+          </span>
+          {asc && <i className='pi pi-arrow-up' style={styles.sortIcon} />}
+          {desc && <i className='pi pi-arrow-down' style={styles.sortIcon} />}
+        </React.Fragment>;
+
+        return <Column
+          style={styles.tableBody}
+          bodyStyle={styles.columnBody}
+          key={col.name}
+          field={col.name}
+          header={header}
+          sortable={true} />;
+      });
+
+      return <div style={{position: 'relative'}}>
+        <style>{css}</style>
+        {data && <DataTable
+          style={styles.table}
+          ref={(el) => this.dt = el}
+          value={data}
+          sortField={sortField}
+          sortOrder={sortOrder}
+          onSort={this.onSort}
+          paginator={true}
+          paginatorTemplate={data.length ? paginatorTemplate : ''}
+          currentPageReportTemplate={data.length ? pageReportTemplate : ''}
+          onPage={this.onPage}
+          first={start}
+          rows={rows}
+          totalRecords={data.length}
+          scrollable={true}
+          scrollHeight='calc(100vh - 350px)'
+          autoLayout={true}
+          emptyMessage=''>
+          {columns}
+        </DataTable>}
```

Comment:
If there is no data we need to show some error messages and should not allow user to sort columns.

Reply:
Added a message for no results. Do we need to worry about sorting if there's no data?